core/src/main/java/org/bitcoinj/core/MessageSerializer.java

```diff
@@ -39,7 +39,9 @@ public int getProtocolVersion() {
     }
 
     public void setProtocolVersion(int protocolVersion) {
-        this.protocolVersion = protocolVersion;
+        this.protocolVersion = Math.min(protocolVersion, NetworkParameters.ProtocolVersion.CURRENT.getBitcoinProtocolVersion());
+        // the version we select should >= to MINIMUM
+        this.protocolVersion = Math.min(this.protocolVersion, NetworkParameters.ProtocolVersion.MINIMUM.getBitcoinProtocolVersion());
```

Comment:
Shouldn't the second check be `Math.max` then? With the current check, we always end up with `MINIMUM` as the protocol version, even if `CURRENT` is bigger.

If this is intended, perhaps the comment can be modified to be more clear.